resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -2,7 +2,7 @@
 jruby-puppet: {
     # Where the puppet-agent dependency places puppet, facter, etc...
     # Puppet server expects to load Puppet from this location
-    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby, /opt/puppetlabs/puppet/cache/lib]
```

Comment:
@camlow325 Prior to 4.0, the master (webrick, passenger, puppetserver) and agent shared the same libdir (`/var/lib/puppet/lib`). And yep, we did have this issue in the past (which is why I filed https://projects.puppetlabs.com/issues/18154). So I agree with you,  the current patch is no worse than it used to be.